src/libs/ComposerFocusManager.ts

```diff
@@ -1,25 +1,273 @@
-let isReadyToFocusPromise = Promise.resolve();
-let resolveIsReadyToFocus: (value: void | PromiseLike<void>) => void;
+import type {View} from 'react-native';
+import {TextInput} from 'react-native';
+import type {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import isWindowReadyToFocus from './isWindowReadyToFocus';
 
-function resetReadyToFocus() {
-    isReadyToFocusPromise = new Promise((resolve) => {
-        resolveIsReadyToFocus = resolve;
+type ModalId = number | undefined;
+
+type InputElement = (TextInput & HTMLElement) | null;
+
+/**
+ * So far, modern browsers only support the file cancel event in some newer versions
+ * (i.e., Chrome: 113+ / Firefox: 91+ / Safari 16.4+), and there is no standard feature detection method available.
+ * We will introduce this prop to isolate the impact of the file upload modal on the focus stack.
+ */
+type BusinessType = ValueOf<typeof CONST.MODAL.BUSINESS_TYPE> | undefined;
+
+type RestoreFocusType = ValueOf<typeof CONST.MODAL.RESTORE_FOCUS_TYPE> | undefined;
+
+type ModalContainer = View | HTMLElement | undefined | null;
+
+type FocusMapValue = {
+    input: InputElement;
+    businessType?: BusinessType;
+};
+
+type PromiseMapValue = {
+    ready: Promise<void>;
+    resolve: () => void;
+};
+
+let focusedInput: InputElement = null;
+let uniqueModalId = 1;
+const focusMap = new Map<ModalId, FocusMapValue>();
+const activeModals: ModalId[] = [];
```

Comment:
>We still need to get the index of the modalId in the queue, and the last element in the queue. not sure, could Set conveniently fulfill this requirement? 🤔

Interesting, will have a deeper look, i thought it would be the most suitable